Xamarin.PropertyEditing.Tests/MockObjectEditor.cs

```diff
@@ -291,6 +312,25 @@ public Task<AssignableTypesResult> GetAssignableTypesAsync (IPropertyInfo proper
 				Value = default(T)
 			};
 		}
+
+		public Task<ITypeInfo> GetValueTypeAsync (IPropertyInfo property, PropertyVariation variation = null)
+		{
+			if (variation != null)
+				throw new NotImplementedException();
+
+			Type type = property.Type;
+			if (this.values.TryGetValue (property, out object value)) {
```

Comment:
We could easily avoid the reflection here by having a `Type Type => typeof(T);` property on `ValueType<T>`, and then doing `if (value is ValueInfo vi) { value = vi.Value; type = vi.Type; }`

Reply:
Since something like this could be used to improve more than just mock object, I'm going to file a separate issue for it, but it's a good idea.